### PR TITLE
Janitorial: Don't use bind in on* handlers

### PR DESCRIPTION
**Pull request by aidvu**
Create a new component and pass the props so we're not binding handlers.

Test:
- Go to Domain Management
- Click Email tab on top
- Try adding a Google Apps subscription
- Try adding multiple subscriptions for different domains
- Make sure all tracking events are fired
- Make sure checkout finished successfully

### CODE REVIEW

File: client/components/upgrades/google-apps/google-apps-dialog/users-form.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		);
+
+		return (
+			<div className="google-apps-dialog__user-fields" key={ `google-apps-dialog-user-${ index }` }>
+				<FormFieldset>
+					<FormTextInputWithAffixes
+						className={ this.fieldClasses( 'email' ) }
+						placeholder={ translate( 'e.g. %(example)s', { args: { example: contactText } } ) }
+						name="email"
+						value={ user.email.value }
```
Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/users-form.jsx]: Unless we specify that an object is required, note that it's generally a good idea to guard against multiple property accessors (`foo.bar.baz`) by checking for a truthy value or using lodash `get`

A common error is that an object isn't available yet (`undefined` or `null`), which may lead to a js error.

**Comment by gwwar** [comment]
Overall the code looks reasonable, we'll want to smoke test before 🚢

File: client/components/upgrades/google-apps/google-apps-dialog/users-form.jsx (hunk 1 above)
Comment (aidvu) [inline client/components/upgrades/google-apps/google-apps-dialog/users-form.jsx]: This is mostly a port of the old code. Tho I guess you're right, `get` with a default should work better. :+1:

Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/users-form.jsx]: It's optional, but I just wanted to note it. 😄

Comment (aidvu) [inline client/components/upgrades/google-apps/google-apps-dialog/users-form.jsx]: No, no, it's a good point! When I'm doing janitorial should take care of these too. :bowing_man:

**Comment by matthusby** [comment]
Code looks good, and works as advertised 👍 

:shipit:

**Pull request merged**